life_expectancy/load_save_data.py

```diff
@@ -1,23 +1,77 @@
 """This model loads and saves the data"""
-from pathlib import Path
+#pylint: disable=R0903
 
+from pathlib import Path
+from typing import Any, Protocol
+import zipfile
 import pandas as pd
 
 
-def load_data(
-        input_file_path: Path
-    ) -> pd.DataFrame:
-    """
-    Load eu_life_expectancy_raw.tsv data from the data folder
-    Args:
-        input_file_path (Path): Path for the file to be loaded
-    Returns:
-        life_exp_raw_data (Pandas DataFrame): DataFrame with data to be cleaned
-    """
-    with open(input_file_path, "r", encoding="UTF-8") as file:
-        life_exp_raw_data = pd.read_csv(file, sep="\t")
+class FileReaderStrategy(Protocol):
+    """Reads the file"""
+
+    def __call__(self, file:Any) -> pd.DataFrame:
+        """
+        Reads the file and returns the data as a pandas DataFrame
+        Args:
+            file (Any): file object to read
+        Returns:
+            pd.DataFrame: The data read from the file as a pandas DataFrame
+        Raises:
+            NotImplementedError: This is a protocol method, so must be implemented by subclasses
+        """
+
+        raise NotImplementedError("This method should be implemented by subclasses")
+
+
+class TSVFileReader:
+    """Reads the TSV File"""
+
+    def __call__(self, file:Any) -> pd.DataFrame:
+        """
+        Reads a TSV file and returns the data as a pandas DataFrame
+        Args:
+            file (Any): file object to read
+        Returns:
+            pd.DataFrame: The data read from the TSV file as a pandas DataFrame
+        """
+
+        return pd.read_csv(file, sep= "\t")
+
+
+class JSONFileReader:
+    """Reads the JSON File"""
+
+    def __call__(self, file:Any) -> pd.DataFrame:
+        """        
+        Reads a JSON file and returns the data as a pandas DataFrame
+        Args:
+            file (Any): file object to read
+        Returns:
+            pd.DataFrame: The data read from the JSON file as a pandas DataFrame"""
+
+        with zipfile.ZipFile(file) as zip_file:
+            with zip_file.open(zip_file.namelist()[0]) as json_file:
+                return pd.read_json(json_file)
+
+
+class FileProcessor:
+    """Processes the file using a FileReaderStrategy"""
+    def __init__(self, file_reader: FileReaderStrategy):
+        self.file_reader = file_reader
+
+    def processor_file(self, file_path: Path):
```

Comment:
```suggestion
    def process_file(self, file_path: Path):
```

Functions names work better as verbs 😉

Reply:
Done :) Thanks!